x-pack/plugins/enterprise_search/public/applications/shared/enterprise_search_provider/index.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { createContext, FC, useContext } from 'react';
+
+import { InitialAppData } from '../../../../common/types';
+
+import { NO_ACCESS } from './constants';
+
+export interface EnterpriseSearchContextData {
+  initialData: InitialAppData;
+}
+const initialContextWithNoAccess: EnterpriseSearchContextData = {
+  initialData: {
+    access: NO_ACCESS,
+  },
+};
+
+const EnterpriseSearchContext = createContext<EnterpriseSearchContextData>(
+  initialContextWithNoAccess
+);
+
+export const EnterpriseSearchProvider: FC<EnterpriseSearchContextData> = ({
```

Comment:
suggestion: For consistency it probably makes more sense to put data this into the Kea KibanaLogic where we already store a lot of Enterprise Search-level data. That way we don't have multiple places where we store app-level concerns. See:
https://github.com/elastic/kibana/blob/4d30305d94b86682caab249915ebf30f9c4e8d74/x-pack/plugins/enterprise_search/public/applications/index.tsx#L51

Reply:
I'm in favor of this. We didn't want to have to create a new logic for each app that needed the access, but if this is already available and shared between the different applications then it makes sense to use it instead of introducing a new Context.

We could also store just the access in this Logic instead of the entire `InitialAppData` object.

Reply:
I've made this change.